pkg/util/kubernetes/apiserver/store.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/patrickmn/go-cache"
+)
+
+// metaBundleStore is a cache for MetadataMapperBundles for each node in the cluster
+// and allows multiple goroutines to safely get or create meta bundles for the same nodes
+// without overwriting each other.
+type metaBundleStore struct {
+	mu      sync.RWMutex
+	cache   *cache.Cache
+	keyFunc func(...string) string
```

Comment:
Is the keyFunc reference necessary ? If it's always going to be BuildAgentKey I don't think that making another layer of abstraction helps readability